### PR TITLE
add Infrastructure team Specs and Interfaces

### LINKED ISSUE #2: add Infrastructure-Team-Lead-Spec
PR creates Infrastructure folder and spec is saved inside Specs sub-folder

### LINKED ISSUE #30: Add Spec - Infrastructure/Logging Framework
Add the spec document for the logging framework
spec will be named membername-spec.md

### LINKED ISSUE #32: Infrastructure Module - Team Member Spec - BadalKumar[111701008]

### LINKED ISSUE #34: Content Module - Team Member Spec Added (111701019)
Talha Yaseen Spec added

### LINKED ISSUE #36: Infrastructure Module - Team Member Spec - Abhishek Saran[111701001]
The team member specs on Test Harness

### LINKED ISSUE #38: Content Module - Adding interface IContentCommunicator

### LINKED ISSUE #40: Add Logger Framework Interface - Stub Implementation
Add a stub implementation of Logger Framework

### LINKED ISSUE #41: Adding IContentNotificationHandler Interface - Talha (111701019)

### LINKED ISSUE #51: Test Harness Interface

### LINKED ISSUE #55: add Infrastructure team Specs and Interfaces
Closed when all specs and interfaces are merged to L1

### PR DESCRIPTION
**Pull request by Rahul-R5**
Initial PR which consists of the design and required interfaces
Closes #55
Closes #51 
Closes #41 
Closes #40 
Closes #30
Closes #34 
Closes #36
Closes #38 
Closes #32
Closes #2

### CODE REVIEW

**Comment by muhammedyaseen-git** [comment]
Why is it still WIP? We are well past the deadline! @Rahul-R5

**Comment by Rahul-R5** [comment]
@convict-git @muhammedyaseen-git Interfaces ready for review

**Comment by convict-git** [comment]
Whenever it's ready, please remove the [WIP] tag. I was just checking all the PRs from outside and had the impression that it's still under WIP. Thanks for the work. We will review soon. :+1:

File: src/infrastructure/validation/logger/ILogger.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * @author Navaneeth M Nambiar
+ */
+public interface ILogger {
+	/**
+	 * 
+	 * @param moduleIdentifier
+	 * @param level
+	 * @param message
+	 */
```
Comment (convict-git) [inline src/infrastructure/validation/logger/ILogger.java]: Comments are not clean and complete.

File: src/infrastructure/validation/logger/LogLevel.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (convict-git) [inline src/infrastructure/validation/logger/LogLevel.java]: Please add some comments. What is the purpose of LogLevel?

File: src/infrastructure/validation/testing/TestCase.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * @author        Abhishek Saran
+ * @module        Infrastructure module 
+ * @team          Test Harness
+ * @description   Abstract class which implements ITest Interface
+ * @summary       Modules have to extend the abstract class to implement the run method for each test case  
+ */
+
+package infrastructure.validation.testing;
+
+public abstract class TestCase implements ITest{
```
Comment (convict-git) [inline src/infrastructure/validation/testing/TestCase.java:11]: Please improve the indentation levels.

File: src/infrastructure/validation/testing/ITest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  void setCategory(String category);
+  void setPriority(int priority);
+
+  /**
+   * abstract method to be implemented for each TestCase
+   * @return  <code>true</code> if the test case pass 
+   *     and actual output is same as expected output; 
+   *     <code>false</code> otherwise.
+   */
+ boolean run();
```
Comment (convict-git) [inline src/infrastructure/validation/testing/ITest.java:29]: Please work on the indentations.

**Comment by convict-git** [comment]
Guys good work. Just need to reflect upon having proper (and consistent) indentations. Also, the logger part looks not very clean. Please add proper comments in the files associated with that.

**Review by convict-git — CHANGES_REQUESTED** [review]

File: src/infrastructure/validation/testing/ITest.java (hunk 4 above)
Comment (abhisheksaran) [reply src/infrastructure/validation/testing/ITest.java:29]: Indentation is already according to the java style document shared by you.

Comment (Rahul-R5) [reply src/infrastructure/validation/testing/ITest.java:29]: I read in the document, if the description for a block tag needs more lines then it is indented using 4 spaces from @. I said the same to abhishek and he followed that.

**Review by convict-git — APPROVED** [review]
Looks great! Thanks :)

**Comment by convict-git** [comment]
@Rahul-R5 Is the PR ready for merge? Please mention in a separate comment "Ready for merge", and also remove the WIP tag.

**Comment by Rahul-R5** [comment]
@convict-git @muhammedyaseen-git  If we get approval from @muhammedyaseen-git , we are ready to merge

**Review by muhammedyaseen-git — APPROVED** [review]
ready to go

**Pull request merged**